### PR TITLE
Skipmap

### PR DESCRIPTION
**Pull request by LuisMartinSchick**
Implemented requested changes by @Thomas-Smyth in #61

### CODE REVIEW

File: plugins/skipmap/index.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
+import { CHAT_MESSAGE, NEW_GAME } from 'squad-server/events';
+import { COPYRIGHT_MESSAGE } from 'core/constants';
+
+export default {
+  name: 'skipmap',
+  description:
+    'The <code>skipmap</code> plugin will allow players to vote via `+/-` if they wish to skip the current map',
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js]: More ` to replace with `<code>...</code>`.

File: plugins/skipmap/index.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+  optionsSpec: {
+    command: {
+      required: true,
+      description: 'The name of the command to be used in chat.',
+      default: '!skipmap'
+    },
+
+    voteDuration: {
+      required: false,
+      description: 'How long the vote should go.',
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js]: "go on for."

File: plugins/skipmap/index.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+      default: 5 * 60 * 1000
+    },
+
+    startTimer: {
+      required: false,
+      description: 'Time to wait before allowing the vote to go through.',
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js]: "Time before voting is allowed." is more consistent with the below.

File: plugins/skipmap/index.js [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      default: 15 * 60 * 1000
+    },
+
+    pastVoteTimer: {
+      required: false,
+      description: 'Time that needs to have passed since the last vote.',
+      default: 10 * 60 * 1000
+    },
+
+    endTimer: {
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js:29]: Maybe move this up one so it appears after the `startTimer` option.

File: plugins/skipmap/index.js [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      if (!voteActive) return;
+
+      // Check if player has voted previously, if yes, remove their vote
+      if (playerVotes[info.steamID]) {
+        if (playerVotes[info.steamID] === '+') votePos--;
+        else voteNeg--;
+      }
+
+      // Record player vote
+      if (playerVotes[info.steamID] === '+') {
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js]: Note for testing: Check whether Squad strips whitespace.

File: plugins/skipmap/index.js [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+    },
+
+    reminderInterval: {
+      required: false,
+      description: 'The time between individual reminders.',
+      default: 2 * 60 * 1000
+    }
+  },
+
+  init: async (server, options) => {
```
Comment (Thomas-Smyth) [inline plugins/skipmap/index.js]: Nothing immediately in this function uses `await`, so no need for `async`.

**Review by Thomas-Smyth — CHANGES_REQUESTED** [review]

File: plugins/skipmap/index.js (hunk 5 above)
Comment (LuisMartinSchick) [reply plugins/skipmap/index.js]: Will add `.trim()` to avoid having to test it.

**Review by Thomas-Smyth — APPROVED** [review]

**Pull request merged**